src/microbe_stage/MembraneShapeGenerator.cs

```diff
@@ -134,6 +134,21 @@ public MembranePointData GenerateShape(IMembraneDataSource parameters)
         return (mesh, surfaceIndex);
     }
 
+    /// <summary>
+    ///   Creates the visual engulf mesh from the overall shape (see GenerateShape method above) that is already created
+    /// </summary>
+    public (ArrayMesh Mesh, int SurfaceIndex) GenerateEngulfMesh(MembranePointData shapeData)
+    {
+        // TODO: Possibly remove this message for this function. This function is also only used graphically.
+        // TODO: should the 3D membrane generation already happen when GenerateMembranePoints is called?
+        // That would reduce the load on the main thread when generating the final visual mesh, though the membrane
+        // properties are also used in non-graphical context (species speed) so that'd result in quite a bit of
+        // unnecessary computations
```

Comment:
I think both of these TODOs can be removed. Like you said you think the triangle strip is very light to generate.